system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -6023,11 +6023,7 @@ class CLdapClient : implements ILdapClient, public CInterface
 
         // set the password.
         Owned<ISecUser> tmpuser = new CLdapSecUser(user->getName(), "");
-        const char* passwd = user->credentials().getPassword();
-        if(passwd == NULL || *passwd == '\0')
-            passwd = "password";
-
-        if (!updateUserPassword(*tmpuser, passwd, NULL))
```

Comment:
What does happen now if user->credentials().getPassword() is empty or nullptr?  It doesn't seem to get checked but may be checked by ldap internally?  What should the behavior be?

Reply:
The LDAP call will fail because we have set the account to require passwords (see line 6015).  However, I went ahead and added a check just to be double secure